packages/codeeditor/src/viewer.ts

```diff
@@ -28,6 +24,19 @@ export class CodeViewerWidget extends Widget {
     layout.addWidget(editorWidget);
   }
 
+  static getCodeViewer(
+    options: CodeViewerWidget.INoModelOptions
+  ): CodeViewerWidget {
+    const model = new CodeEditor.Model({
+      value: options.content,
+      mimeType: options.mimeType
+    });
+    return new CodeViewerWidget({ factory: options.factory, model });
+  }
+
+  getContent = (): string => this.model.value.text;
+  getMimeType = (): string => this.model.mimeType;
```

Comment:
The practice in the code is more to use getter than method for this.
```suggestion
  get content(): string { return this.model.value.text; }
  get mimeType(): string { return this.model.mimeType; }
```

Reply:
will do, I went back and forth on this trying to decide which to use (since both are found in core)